### PR TITLE
Update developer guide

### PR DESCRIPTION
**Pull request by yongping827**
Partially addresses #297.

Changes:
* Add Analytics feature section.
* Add object diagram for Analytics feature section.

### CODE REVIEW

File: docs/DeveloperGuide.md [hunk 1]
```diff
@@ -349,7 +349,28 @@ it to an `Expense` object.
 
 ### Analytics
 
-[Coming soon]
+The Analytics feature allows the user to view their spending and saving trends over a period of time.
```
Comment (ianyong) [inline docs/DeveloperGuide.md:352]: Missing `Overview` subheading.

File: docs/DeveloperGuide.md [hunk 2]
```diff
@@ -349,7 +349,28 @@ it to an `Expense` object.
...
+Based on the current transaction lists, the total monthly expenses, incomes and savings over the past three months are calculated and displayed in the Analytics Tab in the form of bar charts.
+
+#### Implementation of feature
```
Comment (ianyong) [inline docs/DeveloperGuide.md:355]:
```suggestion
##### Implementation of feature
```

File: docs/DeveloperGuide.md [hunk 3]
```diff
@@ -349,7 +349,28 @@ it to an `Expense` object.
...
+1. The method `AnalyticsTabPane#populateData` is called and takes in the `MonthlyBudget` object.
+   The following fields from `MonthlyBudget` are accessed:
+   * `ObservableList<CalculatedAmount>` fields: `monthlyExpenses`, `monthlyIncomes` and `monthlySavings`
+   * `ObservableList<String>` field: `months`
+
+   The method `AnalyticsTabPane#populateDataIn` is called for each `BarChart` object.
+1. In the method `AnalyticsTabPane#populateDataIn`, the values from the above fields in *step 3* are accessed to construct `XY.Data` objects.
+1. The `XY.Data` objects are added to the `BarChart` objects.
+
+#### Design considerations
```
Comment (ianyong) [inline docs/DeveloperGuide.md:371]:
```suggestion
##### Design considerations
```

File: docs/DeveloperGuide.md [hunk 4]
```diff
@@ -349,7 +349,28 @@ it to an `Expense` object.
...
-[Coming soon]
+The Analytics feature allows the user to view their spending and saving trends over a period of time.
+Based on the current transaction lists, the total monthly expenses, incomes and savings over the past three months are calculated and displayed in the Analytics Tab in the form of bar charts.
+
+#### Implementation of feature
+
+![Object diagram for Analytics feature](images/AnalyticsTabObjectDiagram.png)
+
+When the constructor for `AnalyticsTabPane` is called:
+1. `AnalyticsTabPane` constructor takes in a `MonthlyBudget` object from `Model`
```
Comment (ianyong) [inline docs/DeveloperGuide.md:360]:
```suggestion
1. `AnalyticsTabPane` constructor takes in a `MonthlyBudget` object from `Model`.
```

**Review by ianyong — CHANGES_REQUESTED** [review]

**Review by ianyong — APPROVED** [review]
LGTM!

**Review by zhaojj2209 — APPROVED** [review]
LGTM!

**Pull request merged**